src/CMakeLists.txt

```diff
@@ -3,10 +3,19 @@ list(REMOVE_ITEM src_files ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp)
 
 find_package(GSL REQUIRED)
 
+# Fetch the TCLAP library from the git repository and place it
+# into the source folder under src/tclap
+IF(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/tclap )
```

Comment:
In general, I'm not a fan of this approach, for a couple of reasons:

1. It doesn't use built-in CMake functionality to fetch and build tclap, which is designed robustly to handle this exact situation.
2. It modifies the source tree in a way which most users would find surprising and unexpected
3. It fails on my machine (the `git archive` command produces a tar file, not a zip, in defiance of what the documentation says. I needed to add a `--format=zip` to the command.
4. It does not fetch a particular known version of tclap: instead it just gets whatever happens to be on master.
5. It relies on too much knowledge of how the tclap source tree is laid out, which could change.

I think that it's okay for the moment, but once this gets merged in, I'm putting in an issue to switch to using CMake's external project functionality for this.